internal/exec/tar_utils.go

```diff
@@ -0,0 +1,92 @@
+package exec
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	u "github.com/cloudposse/atmos/pkg/utils"
+)
+
+// extractTarball extracts the tarball file into the destination directory
+func extractTarball(sourceFile, extractPath string) error {
+	file, err := os.Open(sourceFile)
+	if err != nil {
+		return err
+	}
+
+	defer func(file *os.File) {
+		err := file.Close()
+		if err != nil {
+			u.LogError(fmt.Errorf("\nerror closing the file '%s': %v\n", sourceFile, err))
+		}
+	}(file)
+
+	var fileReader io.ReadCloser = file
+
+	if strings.HasSuffix(sourceFile, ".gz") {
+		if fileReader, err = gzip.NewReader(file); err != nil {
+			return err
+		}
+
+		defer func(fileReader io.ReadCloser) {
+			err := fileReader.Close()
+			if err != nil {
+				u.LogError(fmt.Errorf("\nerror closing the file '%s': %v\n", sourceFile, err))
+			}
+		}(fileReader)
+	}
+
+	tarBallReader := tar.NewReader(fileReader)
+
+	for {
+		header, err := tarBallReader.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+
+		filename := filepath.Join(extractPath, filepath.FromSlash(header.Name))
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			err = os.MkdirAll(filename, os.FileMode(header.Mode))
+			if err != nil {
+				return err
+			}
+
+		case tar.TypeReg:
```

Comment:
What about unhandled types? It should warn or error. It looks like it just ignores it otherwise.

Reply:
What about symlinks? devices? etc. The tar supports way more than files and directories.

Reply:
I'm not saying we need to handle them, but we shouldn't silently ignore them with zero output.

Reply:
It can be debug level notice. Just don't want it lost.

Reply:
It can be debug level notice. Just don't want it lost.